homeassistant/components/bosch_shc/switch.py

```diff
@@ -0,0 +1,243 @@
+        )
+
+    for switch in session.device_helper.camera_eyes:
+
+        entities.append(
+            CameraEyesSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+            )
+        )
+
+    for switch in session.device_helper.camera_360:
+
+        entities.append(
+            Camera360Switch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+            )
+        )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class SHCSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC switch."""
+
+    def __init__(
+        self,
+        device: SHCDevice,
+        parent_id: str,
+        entry_id: str,
+        description: SwitchEntityDescription,
+    ) -> None:
+        """Initialize a SHC switch."""
+        super().__init__(device, parent_id, entry_id)
+        self.entity_description = description
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return self.entity_description.device_class
+
+    @property
+    def today_energy_kwh(self):
+        """Return the total energy usage in kWh."""
+        return self._device.energyconsumption / 1000.0
+
+    @property
+    def current_power_w(self):
+        """Return the current power usage in W."""
+        return self._device.powerconsumption
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        self._device.state = True
+
+    def turn_off(self, **kwargs):
+        """Turn the switch off."""
+        self._device.state = False
+
+    def toggle(self, **kwargs):
+        """Toggle the switch."""
+        self._device.state = not self.is_on
+
+
+class SHCCameraSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC camera switch."""
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return SwitchDeviceClass.SWITCH
+
+    @property
+    def should_poll(self):
+        """Camera 360 needs polling."""
+        return True
+
+    def update(self):
+        """Trigger an update of the device."""
+        self._device.update()
+
+
+class SmartPlugSwitch(SHCSwitch):
+    """Representation of a SHC smart plug switch."""
+
+    @property
+    def is_on(self):
+        """Return the switch state is currently on or off."""
+        return self._device.state == SHCSmartPlug.PowerSwitchService.State.ON
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "routing": self._device.routing.name,
+        }
+
+
+class LightSwitch(SHCSwitch):
+    """Representation of a SHC light switch."""
+
+    @property
+    def is_on(self):
+        """Return the switch state is currently on or off."""
+        return self._device.state == SHCLightSwitch.PowerSwitchService.State.ON
```

Comment:
Move `is_on` this to the base class `SHCSwitch`, and you can delete the `LightSwitch` class